### PR TITLE
PRO-94 ensure disabled fields

### PR DESCRIPTION
**Pull request by stuartromanek**
- `readOnly`'ify everything
- correct disabled states and slat behavior for Relationships
- Separate `disabled` from `limitReached` concepts in Attachment fields to preserve correct edit states
- provide `field` to areas so they can be disabled in editors
- Ability to disable `AposContextMenu`s from the top level props
- Provide generic `disabled` tooltip message as computed property from the inputMixin
- be able to disable `range`, `color`, `select`, `radio`
- Various style updates to slats, buttons

### CODE REVIEW

File: modules/@apostrophecms/schema/index.js [hunk 1]
```diff
@@ -1828,6 +1828,7 @@ module.exports = {
               await self.apos.util.recursionGuard(req, `${_relationship.type}:${_relationship.withType}`, () => {
                 // Allow options to the getter to be specified in the schema,
                 // notably editable: true
+                // TODO editable going away in favor of disabled for consistency
```
Comment (stuartromanek) [inline modules/@apostrophecms/schema/index.js:1831]: @boutell I found this note while normalizing relationship field's disabled states, I don't think we use this `editable` property anymore, is there anything to be done here?

File: modules/@apostrophecms/schema/ui/apos/mixins/AposInputMixin.js [hunk 2]
```diff
@@ -76,6 +76,14 @@ export default {
         return 20;
       }
     },
+    tooltip () {
+      let msg = false;
+      if (this.field.readOnly) {
+        // TODO For editor-facing tooltips is this the correct language?
```
Comment (stuartromanek) [inline modules/@apostrophecms/schema/ui/apos/mixins/AposInputMixin.js:82]: Not sure if there should be any more nuance to the generic tooltip message. Disabled? Read Only?

File: modules/@apostrophecms/schema/index.js (hunk 1 above)
Comment (boutell) [reply modules/@apostrophecms/schema/index.js:1831]: It's dead, you can drop the note.

File: modules/@apostrophecms/schema/index.js [hunk 3]
```diff
@@ -1875,6 +1876,7 @@ module.exports = {
 
           // Allow options to the getter to be specified in the schema,
           // notably editable: true
+          // TODO editable going away in favor of disabled for consistency
```
[the remark this answers is not in the document]
Comment (boutell) [reply modules/@apostrophecms/schema/index.js:1879]: You can drop the TODO.

File: modules/@apostrophecms/schema/ui/apos/mixins/AposInputMixin.js (hunk 2 above)
Comment (boutell) [reply modules/@apostrophecms/schema/ui/apos/mixins/AposInputMixin.js:82]: 🤷‍♂️

**Review by boutell — DISMISSED** [review]
I answered your Q's about those TODOs but they aren't what you came to do so I approved as well. This was clearly a Jorb, so thanks.

**Comment by stuartromanek** [comment]
if you could ring the bell again, post-comment cleanup

**Review by boutell — APPROVED** [review]

**Pull request merged**